src/ctest.ts

```diff
@@ -659,26 +684,27 @@ export class CTestDriver implements vscode.Disposable {
                             // Use RegEx to match the format "file_path:line" in value[0]
                             const match = defSourceLineProperty.value.match(/(.*):(\d+)/);
                             if (match && match[1] && match[2]) {
-                                const testDefFile = match[1];
-                                const testDefLine = parseInt(match[2]);
-                                if (!isNaN(testDefLine)) {
-                                    testItem = initializedTestExplorer.createTestItem(test.name, test.name, vscode.Uri.file(testDefFile));
-                                    testItem.range = new vscode.Range(new vscode.Position(testDefLine - 1, 0), new vscode.Position(testDefLine - 1, 0));
+                                testDefFile = match[1];
+                                testDefLine = parseInt(match[2]);
+                                if (isNaN(testDefLine)) {
+                                    testDefLine = undefined;
+                                    testDefFile = undefined
                                 }
                             }
                         }
-                        if (!testItem) {
+
+                        if (!testDefFile) {
                             // Use the backtrace graph to find the file and line number
                             // This finds the CMake module's file and line number and not the test file and line number
-                            const testDefFile = this.tests!.backtraceGraph.files[this.tests!.backtraceGraph.nodes[test.backtrace].file];
-                            const testDefLine = this.tests!.backtraceGraph.nodes[test.backtrace].line;
-                            testItem = initializedTestExplorer.createTestItem(test.name, test.name, vscode.Uri.file(testDefFile));
-                            if (testDefLine !== undefined) {
-                                testItem.range = new vscode.Range(new vscode.Position(testDefLine - 1, 0), new vscode.Position(testDefLine - 1, 0));
-                            }
+                            testDefFile = this.tests!.backtraceGraph.files[this.tests!.backtraceGraph.nodes[test.backtrace].file];
+                            testDefLine = this.tests!.backtraceGraph.nodes[test.backtrace].line;
                         }
-                    } else {
-                        testItem = initializedTestExplorer.createTestItem(test.name, test.name);
+                    }
+
+                    const { testItem, parentSuiteItem } = this.createTestItemAndSuiteTree(test.name, testExplorerRoot, initializedTestExplorer,  testDefFile ? vscode.Uri.file(testDefFile) : undefined);
+
+                    if (testDefLine !== undefined) {
+                        testItem.range = new vscode.Range(new vscode.Position(testDefLine - 1, 0), new vscode.Position(testDefLine - 1, 0));
```

Comment:
Minor: it would make sense for both the file and line to be set either here or in `createTestItemAndSuiteTree` (in the same place). Both can be set by property assignments for uniformity.

Reply:
To reduce regression risk, I'd prefer to keep this as is for now.